dali/test/python/test_operator_nonsilence.py

```diff
@@ -0,0 +1,103 @@
+    length = index[1] - index[0]
+    return np.array([begin]), np.array([length])
+
+
+class NonsilencePipeline(Pipeline):
+    def __init__(self, batch_size, num_threads=1, exec_async=True, exec_pipelined=True):
+        super(NonsilencePipeline, self).__init__(batch_size, num_threads, 0, seed=42,
+                                                 exec_async=exec_async,
+                                                 exec_pipelined=exec_pipelined)
+        self.input = ops.FileReader(device="cpu", file_root=audio_files)
+        self.decode = ops.AudioDecoder(device="cpu", dtype=types.FLOAT, downmix=True)
+
+        self.nonsilence = None
+
+    def define_graph(self):
+        if self.nonsilence is None:
+            raise RuntimeError(
+                "Error: you need to derive from this class and define `self.nonsilence` operator")
+        read, _ = self.input()
+        audio, rate = self.decode(read)
+        begin, len = self.nonsilence(audio)
+        return begin, len
+
+
+class NonsilenceDaliPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_power,
+                 reset_interval):
+        super(NonsilenceDaliPipeline, self).__init__(batch_size, num_threads=1)
+        self.nonsilence = ops.NonsilentRegion(cutoff_db=cutoff_value, window_length=window_size,
+                                              reference_power=reference_power,
+                                              reset_interval=reset_interval,
+                                              device="cpu")
+
+
+class NonsilenceRosaPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_power, reset_interval):
+        super(NonsilenceRosaPipeline, self).__init__(batch_size, num_threads=1,
+                                                     exec_async=False, exec_pipelined=False)
+        hop_length = 1
+        function = partial(trim_ref, cutoff_value,
+                           np.max if reference_power == 0. else reference_power,
+                           window_size, hop_length)
+        self.nonsilence = ops.PythonFunction(function=function, num_outputs=2)
+
+
+def check_nonsilence_operator(batch_size, cutoff_value, window_size, reference_db,
+                              reset_interval, eps):
+    test_utils.compare_pipelines(
+        NonsilenceDaliPipeline(batch_size, cutoff_value, window_size, reference_db, reset_interval),
+        NonsilenceRosaPipeline(batch_size, -cutoff_value, window_size, reference_db,
+                               reset_interval), batch_size=batch_size, N_iterations=1, eps=eps)
+
+
+def test_nonsilence_operator():
+    batch_size = 1
+    eps = 1
+    window_sizes = [512, 1024, 2048]
+    window_size_to_reset_interval = [3, 4, 5]
+    reset_intervals = [-1] + list(
+        map(lambda x: x[0] * x[1], itertools.product(window_sizes, window_size_to_reset_interval)))
+    references_power = [0.]
```

Comment:
```suggestion
    references_power = [None,]
```
how about also testing a proper value?

Reply:
Will `None` be properly converted to float `0.`? Because that's the default value in the operator

Reply:
The default value 0. should not be explicitly checked. You should check if `HasArgument` and then decide what to do. `0.` is a bad value because it causes division by cero if it was actually used (therefore confusing)

Reply:
Done